ambry-server/src/main/java/com.github.ambry.server/AmbryRequests.java

```diff
@@ -638,12 +705,17 @@ private void sendGetResponse(RequestResponseChannel requestResponseChannel, GetR
   /**
    * Check that the provided partition is valid, on the disk, and can be written to.
    * @param partition the partition to validate.
-   * @param checkPartitionState {@code true} to check if a partition is read-write.
+   * @param requestType the {@link RequestOrResponseType} being validated.
    * @return {@link ServerErrorCode#No_Error} error if the partition can be written to, or the corresponding error code
    *         if it cannot.
    */
-  private ServerErrorCode validateRequest(PartitionId partition, boolean checkPartitionState) {
-    // 1. check if partition exists on this node and that the store for this partition has been started
+  private ServerErrorCode validateRequest(PartitionId partition, RequestOrResponseType requestType) {
+    // 1. Check partition is null
+    if (partition == null) {
+      metrics.badRequestError.inc();
+      return ServerErrorCode.Bad_Request;
```

Comment:
bad request does not mean malformed request - it just means that the request is bad (either because of having bad params or being unreadable). In this case, I debated b/w `Partition_Unknown` v/s this and thought this was the better one since the requestor has not specified a partition.